File: Modules/UserRoles/UserRolesForm.cs

namespace MyProject.Forms
{
    using Serenity;
    using Serenity.ComponentModel;
    using Serenity.Data;
    using System;
    using System.ComponentModel;
    using System.Collections.Generic;
    using System.IO;

    [FormScript("UserRoles")]
    [BasedOnRow(typeof(Entities.UserRolesRow), CheckNames = true)]
    public class UserRolesForm
    {
        public Int32 UserId { get; set; }
        public Int32 RoleId { get; set; }
    }
}

File: Imports/ServerTypings/UserRolesForm.ts

namespace MyProject {
    export class UserRolesForm extends Serenity.PrefixedContext {
        static formKey = 'UserRoles';
    }

    export interface UserRolesForm {
        UserId: Serenity.IntegerEditor;
        RoleId: Serenity.IntegerEditor;
    }

    [,
        ['UserId', () => Serenity.IntegerEditor],
        ['RoleId', () => Serenity.IntegerEditor]
    ].forEach(x => Object.defineProperty(UserRolesForm.prototype, <string>x[0], {
        get: function () {
            return this.w(x[0], (x[1] as any)());
        },
        enumerable: true,
        configurable: true
    }));
}